### PR TITLE
ClipIt Clipboard Manager History Access

### PR DESCRIPTION
**Pull request by AXGKl**
ClipIt is another popular Clipboard manager in Linux.

This scans the historyfile and puts selected matches into the clipboard

### CODE REVIEW

**Comment by hulloanson** [comment]
A <SPACE> trigger LOL... do you really mean it? To me it gets too easily triggered...

**Comment by AXGKl** [comment]
:-) 
Invoking albert itself is not done by that trigger, its only the prefix to launch the clipboard access within albert's search box.

I need the clipboard history a lot and why would I start anything else albert can launch with a space key (?)

**Comment by hulloanson** [comment]
ok I see... just thought that a \<SPACE\> prefix isn't very indicative in case my thumb glitches and I don't notice it

**Comment by AXGKl** [comment]
Yep - definitely not the right prefix for the 'send email to boss' action.

On Alfred on my MBP there is the feature to invoke the launcher itself PLUS a specific action prefix with ONE hotkey.
Could not check if Alfred has that, so thats why I kept the prefix so short - was on graphical Linux just a few days while my machine was in repair and gave Linux a run for the boot OS. Now back on Linux in containers only, so I can't check or fix, sorry....

**Comment by ManuelSchneid3r** [comment]
Yes space is not allowed. One one hand i use space to bypass triggers one the other this is the trigger we all like to have on our favorite extension.

File: ClipIt [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+__iid__ = 'PythonInterface/v0.1'
+__prettyname__ = 'Clipit'
+__version__ = '1.0'
+__trigger__ = ' ' # add your own prefix here. I use <SPACE> key.
+__author__ = 'Gunther Klessinger'
+__dependencies__ = []
+
+fn_hist = '%(HOME)s/.local/share/clipit/history' % os.environ
+max_matches = 10 # more albert won't show anyway
```
Comment (ManuelSchneid3r) [inline ClipIt]: > more albert won't show anyway?
Huh? Actually it should show everything

File: ClipIt [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        return hist[1]
+    hist[0] = ts
+    hist[1] = H = [] # all entries
+
+    def add(lines, H):
+        s = ''.join(lines)
+        if s.strip() and not s in H:
+            H.append(s)
+
+    with open(fn_hist,'rb') as f:
```
Comment (ManuelSchneid3r) [inline ClipIt]: A lot of magic numbers here. What is this all about?

File: ClipIt [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                    return H
+    add(lines, H) if lines else 0
+    return H
+
+
+def make_item(lines, query, i):
+    '''One item per found match in the history'''
+    l = lines.strip().split('\n')
+    res, lens = '', ''
+    if len(l) > 1:
```
Comment (ManuelSchneid3r) [inline ClipIt]: Text elision is done by albert implicitly. Just use the full text.

File: ClipIt [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+
+def handleQuery(query):
+    '''api entry'''
+    if not query.isTriggered:
+        return
+
+    stripped = query.string.strip()
+
+    if not stripped:
```
Comment (ManuelSchneid3r) [inline ClipIt]: Just show the full MRU items if no filter is given.

**Comment by ManuelSchneid3r** [comment]
There is a new [community repository for python extensions](https://github.com/albertlauncher/python_community) now. Please send the PR there. Sorry for the inconvenience. I am just trying to reduce the workload here and make you happy with contemporary merges. If you want to join the party (write access) ask for it in the community chats.

**Pull request closed without merging**